golemfactory/yagna-service-erigon
# Modify the Erigolem runtime for multiactivity & erigon run separately

This PR resolves following JIRA ticket

 
- [APPS-235](https://golemproject.atlassian.net/browse/APPS-235)
- [APPS-241](https://golemproject.atlassian.net/browse/APPS-241)


## Description



### Why
- Erigon should be running and syncing independently of the demand on Golem Network.

### What
- manages access: adds & removes users
- on activity startup checks if Erigon works (doesn’t start/stop it) by making a request to Erigon RPC


## Testing instructions
Remember to change config `$HOME/.local/share/ya-runtime-erigon/ya-runtime-erigon.json`
```json
{  
  "erigon_http_addr": "<WORKING ERIGON RPC HOST HERE>",
  "erigon_http_port": "8545",
  "passwd_tool_path": "htpasswd",
  "passwd_file_path": "<PATH TO PASS_FILE FOR TESTS>",

  // ...
}

```

Check runtime in the debugger
```bash
RUST_BACKTRACE=1 ya-runtime-dbg \  
  --runtime ./target/debug/ya-runtime-erigon \
  --workdir ./target/debug
```

### Verify that:
- when the runtime starts, new user `erigolem_PID` is created
- check credentials: `htpasswd -bv <passfile> erigolem_PID <password>`
- when the runtime terminates user is deleted
- if the passfile does not exists runtime creates it
- if there is other user than `erigolem_PID` it won't be deleted after runtime terminates
- runtime fails to start when erigon rpc is not awailable
- runtime fails to start when erigon rpc responds with unexpected response


### Assumptions 

[-] _?_

### How to test

[-] _?_
